### PR TITLE
Create SEO component to easily set title / meta information

### PR DESCRIPTION
**Pull request by Limeoats**
Created an SEO component to easily set the page title / meta information for each page in the application.

## What type of Pull Request is this?
**Delete all options except for the one that applies**
- Feature

## What is the current behavior?

You would have to manually work with the `<svelte:head>` on each page to set a title.

Issue Number: N/A

## What is the new behavior?

- Include the SEO component and set the props for `title` and `description`
- More props can be added in the future
- Additionally, more functionality can be added to the SEO component in one spot that can then be injected onto each page (analytics tags, more meta stuff, etc.)

### CODE REVIEW

**Comment by omar2205** [comment]
Can you turn it into a store? Then we can update the store in pages (or components if we need to)

**Comment by w3cj** [comment]
> Can you turn it into a store? Then we can update the store in pages (or components if we need to)

Yeah I second this. I think for certain pages it will make sense to import the `Seo` component and specify the props directly, but for convenience it will be nice to just import a store and set the title / description there.

**Comment by Limeoats** [comment]
So for the title and description to be injected into the `<head>` of the page, it needs to be set within the `<svelte:head>` tag. Sure, we can store the page title / description in a store for easier updating, but it still needs to be put into that `<svelte:head>` somewhere. That's where the `Seo` component comes in since you just pop that onto a page and it handles it for you.

The `Seo` component could pull the title and description from the store and set them that way (or through props), but you'd still need to put the `Seo` component onto each page either way.

I'm fairly new to Svelte so I might be missing something!

**Comment by Limeoats** [comment]
So I think I was able to accomplish the best of both methods with my latest commit. The `title` and `description` are now in a store, so those values can be updated from any component or page. I put an instance of the `Seo` component on the base layout so it'll automatically pull in those values. However, if you want to, you can still use the `Seo` component directly and set the props.

I used both methods to show them working on the two existing pages in the app.

**Comment by paoloricciuti** [comment]
> You will just put the `<Seo />` or `<svelte:head>` in the main root layout.
> 
> ```ts
> // src/lib/SeoStore.ts or // src/lib/MainStore.ts
> import { writable } from 'svelte/store'
> 
> export const Seo = writable({
>   title: 'Home'
> })
> ```
> 
> ```svelte
> 
> <svelte:head>
>   <title>{$Seo.title || 'Home Page'}</title>
> </svelte:head>
> ```
> 
> > Honestly tho I feel like the SEO component would be clearer and less bug prone.
> 
> Why do you think this is prone to bugs?

Maybe it's just me but given that the SEO tags are part of the markup I would prefer to have them as a component.

But now that I think about it you can't have more than one svelte:head inside a page so if you want to update it inside a layout and than override it in a page you couldn't do it with the component but it works fine with the store.

**Comment by omar2205** [comment]
@Limeoats Please add/update these tags. `og:image` is a placeholder until CJ decides listd's branding.

```html
<title>listd - Create, share and watch lists of YouTube channels</title>
<meta name="description" content="listd is a Full Stack App allowing users to create, share and watch lists of YouTube channels. Whether you want to discover new content, organize your favorites or share your opinions, listd is the app for you." />
<meta name="keywords" content="listd, youtube, channels, lists, app" />

<meta property="og:title" content="listd - Create, share and watch lists of YouTube channels" />
<meta property="og:description" content="listd is a Full Stack App allowing users to create, share and watch lists of YouTube channels. Whether you want to discover new content, organize your favorites or share your opinions, listd is the app for you." />
<meta property="og:image" content="https://user-images.githubusercontent.com/1373867/221797903-42b3a12d-7d63-4284-b17a-83e2721a5d83.png" />
<meta property="og:url" content="https://listd.me" />
```

Obviously, duplicated items share the same store value

**Comment by Limeoats** [comment]
Is it best practice to store all of that data in the store? Most of that will never be modified by other components / pages so I would think it would be hardcoded into the `Seo` component, and then `description`, `title`, `og:title`, `og:description`, `twitter:title`, and `twitter:description` would pull the data from the store (sharing the same data as you mentioned). Do you agree?

**Comment by omar2205** [comment]
@Limeoats We will need to update these with a list’s image, title, and description in the future. You can just hard code them for now and maybe add a note/todo.

**Comment by Limeoats** [comment]
Ok, I added the extra fields. More can be included in the future by just adding them to the Seo component.

**Comment by omar2205** [comment]
We will also need to make some pages pre-renderable (`export const prerender = true` in `+server.ts`) to get these meta tags in the HTML. Or not use a store...

File: src/routes/SEO.svelte [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<script lang="ts">
+	import { seo } from '$lib/stores/SeoStore';
+
+	export let title = 'listd';
+	export let description = 'listd';
+
+	seo.set({
+		title,
+		description,
+	});
```
Comment (omar2205) [inline src/routes/SEO.svelte:10]: When this component mounts, it resets the store (server-side store vs client-side store, i think), re"set" it in `onMount` call.
```ts
function loadSeo() {
	seo.set({
		title,
		description,
	})
}

// First time
loadSeo()

// Second time
onMount(() => {
  loadSeo()
})
```

**Review by omar2205 — CHANGES_REQUESTED** [review]

File: src/routes/SEO.svelte [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<script lang="ts">
+	import { seo } from '$lib/stores/SeoStore';
+
+	export let title = 'listd';
```
Comment (omar2205) [inline src/routes/SEO.svelte:4]: Change to
```ts
export const let title = $seo.title
export const let description = $seo.description
```

It already has sane defaults in the store file.

**Review by omar2205 — CHANGES_REQUESTED** [review]

File: src/routes/SEO.svelte (hunk 1 above)
Comment (Limeoats) [reply src/routes/SEO.svelte:19]: Good call, updated.

File: src/routes/SEO.svelte (hunk 2 above)
Comment (Limeoats) [reply src/routes/SEO.svelte]: Updated

File: src/routes/SEO.svelte [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	<title>{title}</title>
+	<meta name="description" content={description} />
+
+	<meta name="keywords" content="listd, youtube, channels, lists, app" />
+
+	<meta property="og:title" content={title} />
+	<meta property="og:description" content={description} />
+	<meta
+		property="og:image"
+		content="https://user-images.githubusercontent.com/1373867/221797903-42b3a12d-7d63-4284-b17a-83e2721a5d83.png"
```
Comment (w3cj) [inline src/routes/SEO.svelte:32]: Lets remove images until we settle on a logo. Otherwise lgtm!

**Review by w3cj — CHANGES_REQUESTED** [review]

File: src/routes/SEO.svelte [hunk 4]
```diff
@@ -27,21 +27,9 @@
 
 	<meta property="og:title" content={title} />
 	<meta property="og:description" content={description} />
-	<meta
```
Comment (omar2205) [inline src/routes/SEO.svelte]: could've used an empty string...

**Review by omar2205 — CHANGES_REQUESTED** [review]

**Comment by omar2205** [comment]
lgtm

**Comment by omar2205** [comment]
finally the \<Seo /> saga is merged.

**Pull request merged**